fruit-fly/hyperparam_search.py

```diff
@@ -175,9 +175,13 @@ def _classify(topword, KC_size, proj_size, percent_hash, C):
     vocab, reverse_vocab, logprobs = read_vocab()
     PN_size = len(vocab)
     vectorizer = CountVectorizer(vocabulary=vocab, lowercase=False, token_pattern='[^ ]+')
-    train_set, train_label = read_n_encode_dataset(train_path)
-    val_set, val_label = read_n_encode_dataset(train_path.replace('train', 'val'))
+    print('reading dataset')
+    train_set, train_label = read_n_encode_dataset(train_path, vectorizer, logprobs)
+    val_set, val_label = read_n_encode_dataset(train_path.replace('train', 'val'), vectorizer, logprobs)
     max_val_score = -1
+    num_threads = 5
+    while num_threads > multiprocessing.cpu_count() - 1:
+        num_threads = num_threads // 2
```

Comment:
I would have expected something like:

     #Let's imagine I want 5 threads but only have 2 CPUs:
     num_cpus = multiprocessing.cpu_count()    #2
     num_iter = num_threads // num_cpus          #2

     #Do 2 lots of processing
     for i in range(num_iter):
          optimize_fruitfly(continue_log, num_cpus)
    
     #Check the remainder: 5 % 2 = 1
     if num_threads % num_cpus != 0:
           optimize_fruitfly(continue_log, num_threads % num_cpus)    #Process 1 more thread